### PR TITLE
feat: Shopping list UI overhaul - collapsible labels

### PR DESCRIPTION
**Pull request by Wetzel402**
## What type of PR is this?

- feature

## What this PR does / why we need it:

- Make labels in the shopping list collapsible. 

## Which issue(s) this PR fixes:

https://github.com/mealie-recipes/mealie/discussions/4175

## Testing

Tested in dev container

### CODE REVIEW

File: frontend/pages/shopping-lists/_id.vue [hunk 1]
```diff
@@ -57,18 +57,21 @@
...
-              }"
-            >
-              {{ key }}
-          </v-btn>
-          </div>
-          <v-divider/>
+          <span>
+            <div class="text-left">
+              <button @click="toggleShowLabel(key)">
+                <v-btn :color="getLabelColor(value[0]) ? getLabelColor(value[0]) : '#959595'">
```
Comment (Kuchenpirat) [inline frontend/pages/shopping-lists/_id.vue]: any reason you removed the styles?

File: frontend/pages/shopping-lists/_id.vue [hunk 2]
```diff
@@ -57,18 +57,21 @@
...
+                  <v-icon>
+                    {{ labelOpenState[key] ? $globals.icons.chevronDown : $globals.icons.chevronRight }}
+                  </v-icon>
+                  {{ key }}
+                </v-btn>
+              </button>
+            </div>
+          </span>
```
Comment (Kuchenpirat) [inline frontend/pages/shopping-lists/_id.vue]: There are some unnecessairy things in here. 

```suggestion
          <v-btn 
            :color="getLabelColor(value[0]) ? getLabelColor(value[0]) : '#959595'"
            @click="toggleShowLabel(key)"
          >
            <v-icon>
              {{ labelOpenState[key] ? $globals.icons.chevronDown : $globals.icons.chevronRight }}
            </v-icon>
            {{ key }}
          </v-btn>
```

**Review by Kuchenpirat — CHANGES_REQUESTED** [review]
First of all thanks for your continued work on the shopping list, your previous changes made it to the highlight section in our v2 release notes, so congrats on that 🚀🥳 

Here are just a few first comments. This is all i have time for today. So this is quite limited,

File: frontend/pages/shopping-lists/_id.vue (hunk 1 above)
Comment (Wetzel402) [reply frontend/pages/shopping-lists/_id.vue]: My mistake.  This is corrected.

File: frontend/pages/shopping-lists/_id.vue [hunk 3]
```diff
@@ -1025,7 +1064,7 @@ export default defineComponent({
       }
 
       // update current user
-      allUsers.value = data.items.sort((a, b) => ((a.fullName || "") < (b.fullName || "") ? -1 : 1));
+      allUsers.value = data.sort((a, b) => ((a.fullName || "") < (b.fullName || "") ? -1 : 1));
       currentUserId.value = shoppingList.value?.userId;
```
Comment (Kuchenpirat) [inline frontend/pages/shopping-lists/_id.vue]: Looks mostly good. Only thing is this, which destroys the settings window. 
Please re-add the items here

**Review by Kuchenpirat — CHANGES_REQUESTED** [review]

File: frontend/pages/shopping-lists/_id.vue (hunk 3 above)
Comment (Wetzel402) [reply frontend/pages/shopping-lists/_id.vue]: Sorry about that.  It should be corrected now.

**Review by Kuchenpirat — APPROVED** [review]

**Pull request merged**